src/Illuminate/Auth/Access/Gate.php

```diff
@@ -356,14 +375,16 @@ protected function callAfterCallbacks($user, $ability, array $arguments, $result
         $arguments = array_merge([$user, $ability, $result], [$arguments]);
 
         foreach ($this->afterCallbacks as $after) {
-            $after(...$arguments);
+            if ($user || $after['options']['guestable']) {
+                $after['callback'](...$arguments);
+            }
```

Comment:
I want to always call it, but just for BC I so any older code you always have a `$user` and not a `null` so you don't get error of using an attribute on a non-object .. just like `before` filter if you want to use it with guests you need to mark it as `guestable`